jib-core/src/main/java/com/google/cloud/tools/jib/plugins/common/ConfigurationPropertyValidator.java

```diff
@@ -73,39 +114,69 @@ public static Authorization getImageAuth(
 
     // Warn if a system property is missing
     if (!Strings.isNullOrEmpty(commandlinePassword) && Strings.isNullOrEmpty(commandlineUsername)) {
-      logger.warn(
+      jibLogger.warn(
           passwordProperty
               + " system property is set, but "
               + usernameProperty
               + " is not; attempting other authentication methods.");
     }
     if (!Strings.isNullOrEmpty(commandlineUsername) && Strings.isNullOrEmpty(commandlinePassword)) {
-      logger.warn(
+      jibLogger.warn(
           usernameProperty
               + " system property is set, but "
               + passwordProperty
               + " is not; attempting other authentication methods.");
     }
 
     // Check auth configuration next; warn if they aren't both set
-    if (Strings.isNullOrEmpty(auth.getUsername()) && Strings.isNullOrEmpty(auth.getPassword())) {
+    if (Strings.isNullOrEmpty(username) && Strings.isNullOrEmpty(password)) {
       return null;
     }
-    if (Strings.isNullOrEmpty(auth.getUsername())) {
-      logger.warn(
-          auth.getUsernamePropertyDescriptor()
+    if (Strings.isNullOrEmpty(username)) {
+      jibLogger.warn(
+          authParameterTemplate.replace("IMAGE", imageParameter).replace("PROPERTY", "username")
```

Comment:
Yeah, I was thinking about this and am gonna use some helper functions instead.

Reply:
I wonder if `String.format()` or `MessageFormat.format()` could work. Not sure the differences or if they have some caveats, but I guess they would work for basic things.

Reply:
Yeah I think I'd like a `String.format()` solution. Using string construction isn't that straightforward here since maven and gradle parameter layouts are just slightly different.